src/main/java/me/jellysquid/mods/lithium/mixin/alloc/vectors/EntityTrackerMixin.java

```diff
@@ -0,0 +1,46 @@
+package me.jellysquid.mods.lithium.mixin.alloc.vectors;
+
+import net.minecraft.server.network.EntityTrackerEntry;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ThreadedAnvilChunkStorage;
+import net.minecraft.util.math.Vec3d;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(ThreadedAnvilChunkStorage.EntityTracker.class)
+public class EntityTrackerMixin {
+
+    @Shadow
+    @Final
+    EntityTrackerEntry entry;
+
+    @Redirect(
+            method = "updateTrackedStatus(Lnet/minecraft/server/network/ServerPlayerEntity;)V",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/util/math/Vec3d;subtract(Lnet/minecraft/util/math/Vec3d;)Lnet/minecraft/util/math/Vec3d;")
+    )
+    private Vec3d removeAllocation(Vec3d playerPos, Vec3d playerLastPos) {
+        return null;
+    }
+
+    @Redirect(
+            method = "updateTrackedStatus(Lnet/minecraft/server/network/ServerPlayerEntity;)V",
+            at = @At(value = "FIELD", target = "Lnet/minecraft/util/math/Vec3d;x:D")
+    )
+    private double calcX(Vec3d vec, ServerPlayerEntity player) {
+        return player.getPos().x - entry.getLastPos().x;
```

Comment:
This patch doubles the calculation to save one allocation, not sure whether this is helpful

Reply:
I removed the vanilla calculation on line 27 by redirecting `Vec3d.subtract()` and returning null.

Reply:
Oh you mean the redirect has 2 variable targets? Wouldn't this get inlined by the JVM?